polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -908,7 +924,9 @@ function IndexTableBase({
       const isCurrentlySorted = index === sortColumnIndex;
       const isPreviouslySorted =
         !isCurrentlySorted && index === lastSortedColumnIndex.current;
-
+      const isRenderAfterSelectEvent =
+        renderAfterSelectEvent.current ||
+        (!hasSelected.current && selectedItemsCount !== 0);
```

Comment:
This line catches the first selection, whether it's the first click event or the first selection after the user interacts with the sortable headers.